src/NumericalAlgorithms/Spectral/SwshDerivatives.hpp

```diff
@@ -0,0 +1,309 @@
+                                                (l + s) * (l - s + 1)));
+}
+
+template <>
+SPECTRE_ALWAYS_INLINE std::complex<double> derivative_factor<Tags::EthbarEth>(
+    int l, int s) noexcept {
+  return static_cast<std::complex<double>>(-(l - s) * (l + s + 1));
+}
+
+template <>
+SPECTRE_ALWAYS_INLINE std::complex<double> derivative_factor<Tags::EthEthbar>(
+    int l, int s) noexcept {
+  return static_cast<std::complex<double>>(-(l + s) * (l - s + 1));
+}
+
+// For a particular derivative represented by `DerivativeKind` and input spin
+// `Spin`, multiplies the derivative spectral factors with
+// `pre_derivative_modes`, returning by pointer via parameter `derivative_modes`
+template <typename DerivativeKind, int Spin>
+void compute_derivative_coefficients(
+    size_t l_max, size_t number_of_radial_points,
+    gsl::not_null<
+        SpinWeighted<ComplexModalVector,
+                     Spin + Tags::derivative_spin_weight<DerivativeKind>>*>
+        derivative_modes,
+    gsl::not_null<SpinWeighted<ComplexModalVector, Spin>*>
+        pre_derivative_modes) noexcept;
+
+// Helper function for dealing with the parameter packs in the utilities which
+// evaluate several spin-weighted derivatives at once. The `apply` function of
+// this struct locates the appropriate mode buffer in the input tuple
+// `pre_derivative_mode_tuple`, and calls `compute_derivative_coefficients`,
+// deriving the coefficients for `DerivativeTag` and returning by pointer.
+template <typename DerivativeTag, typename PreDerivativeTagList>
+struct dispatch_to_compute_derivative_coefficients;
+
+template <typename DerivativeTag, typename... PreDerivativeTags>
+struct dispatch_to_compute_derivative_coefficients<
+    DerivativeTag, tmpl::list<PreDerivativeTags...>> {
+  template <int Spin, typename... ModalTypes>
+  static void apply(
+      const size_t l_max, const size_t number_of_radial_points,
+      const gsl::not_null<SpinWeighted<ComplexModalVector, Spin>*>
+          derivative_modes,
+      const std::tuple<ModalTypes...>& pre_derivative_mode_tuple) noexcept {
+    compute_derivative_coefficients<typename DerivativeTag::derivative_kind>(
+        l_max, number_of_radial_points, derivative_modes,
+        get<tmpl::position_of_first<tmpl::list<PreDerivativeTags...>,
+                                    typename DerivativeTag::derived_from>>(
+            pre_derivative_mode_tuple));
+  }
+};
+
+// Helper function for dealing with the parameter packs in the utilities which
+// evaluate several spin-weighted derivatives at once. The `apply` function of
+// this struct locates the like-spin set of paired modes and nodes, and calls
+// the member function of `SwshTransform` or `InverseSwshTransform` to perform
+// the spin-weighted transform. This function is a friend of both
+// `SwshTransform` and `InverseSwshTransform` to gain access to the private
+// `apply_to_vectors`.
+template <typename Transform, typename FullTagList>
```

Comment:
Clarify the tags in these template arguments.
Also, explain why these are tuples and not tagged tuples.